src/hotspot/share/opto/matcher.hpp

```diff
@@ -325,6 +325,10 @@ class Matcher : public PhaseTransform {
   // should generate this one.
   static const bool match_rule_supported(int opcode);
 
+  // Identify extra cases that we might want to vectorize automatically
+  // And exclude cases which are not profitable to auto-vectorize.
+  static const bool match_rule_supported_vectorization(int opcode, int vlen, BasicType bt);
```

Comment:
May be we should rename it to `match_rule_supported_superword`. That is how we call auto-vectorizer in C2.

Reply:
> May be we should rename it to `match_rule_supported_superword`. That is how we call auto-vectorizer in C2.

Done. 

And update after https://github.com/openjdk/jdk/pull/8877. Thanks.